backend/http_backend/http_backend.go

```diff
@@ -0,0 +1,114 @@
+package httpbackend
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/skroutz/downloader/job"
+)
+
+const DefaultClientTimeout = 30
+
+var (
+	// Based on http.DefaultTransport
+	//
+	// See https://golang.org/pkg/net/http/#RoundTripper
+	transport = &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   10 * time.Second, // was 30 * time.Second
+			KeepAlive: 30 * time.Second,
+			DualStack: true,
+		}).DialContext,
+		MaxIdleConns:          100,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+	}
+
+	clientTimeout time.Duration
+)
+
+// HTTPBackend notifies about a job completion by executing an HTTP request.
+type HTTPBackend struct {
```

Comment:
We could name this as follows: `<package_name>.Backend` and then use them like this: `httpbackend.Backend{}`